File: Difficulty: Easy/The FizzBuzz Program/the-fizz-buzz-program.js
//{ Driver Code Starts
//Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => {
    inputString += inputStdin;
});

process.stdin.on('end', _ => {
    inputString = inputString.trim().split('\n').map(string => {
        return string.trim();
    });
    
    main();    
});

function readLine() {
    return inputString[currentLine++];
}

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for(; i<t; i++) {
        let a = parseInt(readLine());
        let obj = new Solution();
        obj.fizzBuzz(a);
    
console.log("~");
}
}

// } Driver Code Ends

// User function Template for JavaScript

class Solution {
    fizzBuzz(number) {
        // Check if number is divisible by both 3 and 5, return "FizzBuzz"
        // If divisible by 3, return "Fizz"
        // If divisible by 5, return "Buzz"
        // Otherwise, return the number itself
        const result = number % 3 == 0 && number % 5 == 0 ? "FizzBuzz" :
                       number % 3 == 0 ? "Fizz" :
                       number % 5 == 0 ? "Buzz" :
                       number;
        
        // Print the result
        console.log(result);
    }
}

File: Difficulty: Easy/The FizzBuzz Program/the-fizz-buzz-program.java
//{ Driver Code Starts

// } Driver Code Ends

import java.util.*;

public class Main {
    public static void main(String args[]) {
        // Create a scanner object to read input from the user
        Scanner s = new Scanner(System.in);
        
        // Read the integer n from user input
        int n = s.nextInt();
        
        // Check if the number is divisible by both 3 and 5, and print "FizzBuzz"
        if ((n % 3 == 0) && (n % 5 == 0)) {
            System.out.println("FizzBuzz");
        }
        // If the number is divisible by only 5, print "Buzz"
        else if (n % 5 == 0) {
            System.out.println("Buzz");
        }
        // If the number is divisible by only 3, print "Fizz"
        else if (n % 3 == 0) {
            System.out.println("Fizz");
        }
        // If the number is neither divisible by 3 nor 5, print the number itself
        else {
            System.out.println(n);
        }
    }
}



//{ Driver Code Starts.
// } Driver Code Ends